core/inst_handlers/v/RvvFloatInsts.cpp

```diff
@@ -0,0 +1,1287 @@
+            execute(mask_elems.maskBitIterBegin(), mask_elems.maskBitIterEnd());
+        }
+
+        updateFloatCsrs<XLEN>(state);
+
+        return ++action_it;
+    }
+
+    template <typename XLEN, RvvFloatInsts::OperandMode opMode, RvvFloatInsts::FloatFuncs funcs>
+    Action::ItrType RvvFloatInsts::vfUnaryHandler_(atlas::AtlasState* state,
+                                                   Action::ItrType action_it)
+    {
+        VectorConfig* vector_config = state->getVectorConfig();
+
+        switch (vector_config->getSEW())
+        {
+            case 16:
+                if constexpr (funcs.f16 == nullptr)
+                {
+                    sparta_assert(false, "Unsupported SEW value");
+                }
+                else
+                {
+                    return vfUnaryHelper<XLEN, 16, opMode, [](auto src2) {
+                        return func_wrapper(funcs.f16, src2);
+                    }>(state, action_it);
+                }
+                break;
+            case 32:
+                return vfUnaryHelper<XLEN, 32, opMode, [](auto src2)
+                                     { return func_wrapper(funcs.f32, src2); }>(state, action_it);
+                break;
+            case 64:
+                // neither narrowing to 64 bit nor widening from 64 bit
+                if constexpr (opMode.dst != OperandMode::Mode::W
+                              && opMode.src2 != OperandMode::Mode::W)
+                {
+                    return vfUnaryHelper<XLEN, 64, opMode, [](auto src2) {
+                        return func_wrapper(funcs.f64, src2);
+                    }>(state, action_it);
+                }
+            default:
+                sparta_assert(false, "Unsupported SEW value");
+                break;
+        }
+        return ++action_it;
+    }
+
+    template <typename XLEN, size_t elemWidth, RvvFloatInsts::OperandMode opMode, auto func>
+    Action::ItrType vfFloatToIntHelper(atlas::AtlasState* state, Action::ItrType action_it)
+    {
+        using Traits = FuncTraits<decltype(func)>;
+        using ArgType = std::tuple_element_t<0, typename Traits::ArgsTuple>;
+        using IntT = decltype(std::declval<ArgType>().v);
+
+        const AtlasInstPtr & inst = state->getCurrentInst();
+        Elements<
+            Element<opMode.src2 == RvvFloatInsts::OperandMode::Mode::W ? 2 * elemWidth : elemWidth>,
+            false>
+            elems_vs2{state, state->getVectorConfig(), inst->getRs2()};
+        Elements<
+            Element<opMode.dst == RvvFloatInsts::OperandMode::Mode::W ? 2 * elemWidth : elemWidth>,
+            false>
+            elems_vd{state, state->getVectorConfig(), inst->getRd()};
+        softfloat_roundingMode = READ_CSR_REG<XLEN>(state, FRM);
+
+        auto execute = [&]<typename Iterator>(const Iterator & begin, const Iterator & end)
+        {
+            for (auto iter = begin; iter != end; ++iter)
+            {
```

Comment:
Why bother with a copy of begin?
```suggestion
        auto execute = [&](auto iter, const auto & end)
        {
            for (;iter != end; ++iter)
            {
```